src/file_hosting/authorization.rs

```diff
@@ -0,0 +1,66 @@
+use crate::file_hosting::FileHostingError;
+use serde_json::Value;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use base64::encode;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthorizationPermissions {
+    bucket_id: Option<String>,
+    bucket_name: Option<String>,
+    capabilities: Vec<String>,
+    name_prefix: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthorizationData {
+    pub absolute_minimum_part_size: i32,
+    pub account_id: String,
+    pub allowed: AuthorizationPermissions,
+    pub api_url: String,
+    pub authorization_token: String,
+    pub download_url: String,
+    pub recommended_part_size: i32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct UploadUrlData {
+    pub bucket_id: String,
+    pub upload_url: String,
+    pub authorization_token: String,
+}
+
+pub async fn authorize_account(key_id: String, application_key: String) -> Result<AuthorizationData, FileHostingError> {
+    let combined_key = format!("{}:{}", key_id, application_key);
+    let formatted_key = format!("Basic {}", encode(combined_key));
+
+    let req = reqwest::Client::new()
+        .get("https://api.backblazeb2.com/b2api/v2/b2_authorize_account")
+        .header(reqwest::header::CONTENT_TYPE, "application/json")
+        .header(reqwest::header::AUTHORIZATION, formatted_key)
+        .send()
+        .await?;
+
+    let text = &req.text().await?;
+
+    Ok(serde_json::from_str(text)?)
```

Comment:
Reqwest has serde integration, so you can use `Ok(req.json()?)` instead if you add the `"json"` feature.  This is the same will all other instances of this, but I won't comment on them all.

Reply:
+1

Reply:
This method doesn't exist in the async impl of reqwest.

Reply:
Is this not the correct method?  You have to enable the "json" feature, but it's there.  https://docs.rs/reqwest/0.10.6/reqwest/struct.Response.html#method.json

Reply:
Fixed.